### PR TITLE
Move RollingUpdateOnReady to Beta

### PR DESCRIPTION
**Pull request by roberthbailey**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**: Promotes the feature from alpha -> beta.

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by roberthbailey** [comment]
e2e-feature-gates failed with:

```
Error: UPGRADE FAILED: another operation (install/upgrade/rollback) is in progress
```

**Comment by roberthbailey** [comment]
I found https://github.com/helm/helm/issues/4558 which described how to fix this helm error. After a bit of flailing, I figured out that to find the agones chart you have to pass the `-n` flag to select the right namespace (also, using the `-a` flag to see _all_ versions of the chart, even if it is stopped or broken, is useful as well):

```
$ helm list -a -n agones-system
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
agones  agones-system   43              2021-03-29 00:56:41.77927338 +0000 UTC  pending-upgrade agones-1.14.0-dev       1.14.0-dev
```

Then I was able to roll back to the prior version:

```
helm rollback -n agones-system agones 42
Rollback was a success! Happy Helming!
```

Now we can re-test.

**Comment by roberthbailey** [comment]
I found https://github.com/helm/helm/issues/4558 which described how to fix this helm error. After a bit of flailing, I figured out that to find the agones chart you have to pass the `-n` flag to select the right namespace (also, using the `-a` flag to see _all_ versions of the chart, even if it is stopped or broken, is useful as well):

```
$ helm list -a -n agones-system
NAME    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                   APP VERSION
agones  agones-system   43              2021-03-29 00:56:41.77927338 +0000 UTC  pending-upgrade agones-1.14.0-dev       1.14.0-dev
```

Then I was able to roll back to the prior version:

```
helm rollback -n agones-system agones 42
Rollback was a success! Happy Helming!
```

Now we can re-test.

**Comment by roberthbailey** [comment]
I figured out why the tests were failing. I've sent #2035 as a separate PR to verify that the fix works independently of promoting the feature to beta.

**Comment by roberthbailey** [comment]
Different failure:

```
--- FAIL: TestGameServerUnhealthyAfterReadyCrash (309.46s)
    gameserver_test.go:376: Could not get a GameServer ready: waiting for {game-server [{udp-port Dynamic <nil> 7654 0 UDP}] {false 0 0 0}  { 0 0} {{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {[] [] [{game-server gcr.io/agones-images/simple-game-server:0.2 [] []  [] [] [] {map[cpu:{{30 -3} {<nil>} 30m DecimalSI} memory:{{33554432 0} {<nil>}  BinarySI}] map[cpu:{{30 -3} {<nil>} 30m DecimalSI} memory:{{33554432 0} {<nil>}  BinarySI}]} [] [] nil nil nil nil   IfNotPresent nil false false false}] []  <nil> <nil>  map[]   <nil>  false false false <nil> nil []   nil  [] []  <nil> nil [] <nil> <nil> <nil> map[] []}} <nil>} GameServer instance readiness timed out (): waiting for GameServer to be Ready 1617126920/game-serverdxwpz: timed out waiting for the condition
FAIL
```

**Comment by roberthbailey** [comment]
I'm not sure why I'm getting so many flakes... I'll try and reproduce them locally while getting the tests to pass here.

File: site/content/en/docs/Guides/fleet-updates.md [hunk 1]
```diff
@@ -32,6 +32,13 @@ So when a Fleet is edited (any field other than `replicas`, see note below), eit
 1. Shutdown the `maxUnavailable` number of `GameServers` in the Fleet, skipping `Allocated` `GameServers`.
 1. Repeat above steps until all the previous `GameServer` configurations have been `Shutdown` and deleted.
 
+{{% feature publishVersion="1.14.0" %}}
+By default, Fleets will wait for the new GameSevers to become Ready during a Rolling Update, to ensure there is always a set of Ready GameServers before attempting to shut down the previous version Fleet's GameServers.
```
[the remark this answers is not in the document]
Comment (roberthbailey) [reply site/content/en/docs/Guides/fleet-updates.md:36]: I'd considered doing that but I didn't want to make a large callout like we had for alpha. But maybe a note would make sense to make a consistent highlight for beta features.

[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Guides/fleet-updates.md:36]: Yeah, a consistent note would be good I think 👍🏻 given that beta features can change as time goes on, it's probably worth highlighting that to the end user so they can make a decision on adoption.

[the remark this answers is not in the document]
Comment (roberthbailey) [reply site/content/en/docs/Guides/fleet-updates.md]: I've added a shortcode for beta features.

**Review by markmandel — APPROVED** [review]
🥳

**Pull request merged**